apps/battlepass-service/src/index.ts

```diff
@@ -0,0 +1,64 @@
+import { Hono } from 'hono';
+import { cors } from 'hono/cors';
+import { createClient } from '@supabase/supabase-js';
+
+export type Bindings = {
+    HIBISCUS_SUPABASE_SERVICE_KEY: string;
+    NEXT_PUBLIC_HIBISCUS_SUPABASE_API_URL: string;
+};
+
+const HTTP_BAD_REQUEST = 400;
+const INTERNAL_SERVER_ERROR = 500;
+
+const app = new Hono<{ Bindings: Bindings }>();
+
+app.use('/api/*', cors());
+
+app.get('/api/leaderboard', async (c) => {
+    console.log(c.env.NEXT_PUBLIC_HIBISCUS_SUPABASE_API_URL);
+    console.log(c.env.HIBISCUS_SUPABASE_SERVICE_KEY);
+    try {
+        const supabase = createClient(
+            c.env.NEXT_PUBLIC_HIBISCUS_SUPABASE_API_URL,
+            c.env.HIBISCUS_SUPABASE_SERVICE_KEY
+        );      
+
+        const pageNumber = parseInt(c.req.query("pageNumber"));
```

Comment:
nit: Typescript linter complains about these parseInt lines because parseInt needs a string as its argument but the query could be undefined. Could do something like `c.req.query("pageNumber") ?? ""` to get around the linter (i.e. force undefined to be an empty string)